src/services/room-integration/room.controller.service.ts

```diff
@@ -21,12 +23,27 @@ export class RoomControllerService {
     @Inject(WINSTON_MODULE_NEST_PROVIDER) private readonly logger: LoggerService
   ) {}
 
-  public async getRoomCalloutOrFail(roomID: string) {
+  public async getRoomEntityOrFail(roomID: string): Promise<Callout | Post> {
     const room = await this.roomLookupService.getRoomOrFail(roomID, {
-      relations: { callout: { framing: { profile: true } } },
+      relations: {
+        callout: { framing: { profile: true } },
+        post: { profile: true },
+      },
     });
-    return room.callout;
+    const entity = room.callout || room.post;
+    if (!entity) {
+      this.logger.error(
+        `Room with ID ${roomID} does not have a callout or post.`,
+        LogContext.COMMUNICATION
+      );
+      throw new Error(
```

Comment:
- I would suggest whoever consumes `getRoomEntityOrFail` to handle the error.
- Add a jsdoc, indicating what this function `@throws` and when.
This will improve the developer experience and will avoid double or triple logging.
- Avoid using the UUIDs in general. You can use one of: `EntityNotFoundException`, `BaseException`, `BaseExceptionInternal`